seisbench/models/dkpn.py

```diff
@@ -0,0 +1,784 @@
+        self.batch_cfs.append(batch.detach().clone())  # Store WINDOW CFS
+
+        return batch
+
+    def annotate_batch_post(
+            self, batch: torch.Tensor, piggyback=None, argdict=None):
+        # Transpose predictions to correct shape
+        batch = torch.transpose(batch, -1, -2)
+        prenan, postnan = argdict.get(
+            "blinding", self._annotate_args.get("blinding")[1]
+        )
+        if prenan > 0:
+            batch[:, :prenan] = np.nan
+        if postnan > 0:
+            batch[:, -postnan:] = np.nan
+        self.batch_probs.append(batch.detach().clone())
+        return batch
+
+    def extract_cf(self, stream):
+        _prpr = PreProc(**self.default_args)
+        _prpr.work(stream.copy())
+        return _prpr.get_stream()
+
+    def get_cf_windows(self):
+        """ This method extract the single's windows used in CNN process """
+        return self.batch_cfs
+
+    def get_probs_windows(self):
+        """ This method extract the single's windows used in CNN process """
+        return self.batch_probs
+
+    # ================================================================
+    # ==================================  END  PROCESS STREAM
+    # ================================================================
+
+    def classify_aggregate(self, annotations, argdict) -> sbu.ClassifyOutput:
+        """
+        Converts the annotations to discrete thresholds using
+        :py:func:`~seisbench.models.base.WaveformModel.picks_from_annotations`.
+        Trigger onset thresholds for picks are derived from the argdict at keys "[phase]_threshold".
+
+        :param annotations: See description in superclass
+        :param argdict: See description in superclass
+        :return: List of picks
+        """
+        picks = sbu.PickList()
+        for phase in self.labels:
+            if phase == "N":
+                # Don't pick noise
+                continue
+
+            picks += self.picks_from_annotations(
+                annotations.select(channel=f"{self.__class__.__name__}_{phase}"),
+                argdict.get(
+                    f"{phase}_threshold", self._annotate_args.get("*_threshold")[1]
+                ),
+                phase,
+            )
+
+        picks = sbu.PickList(sorted(picks))
+
+        return sbu.ClassifyOutput(self.name, picks=picks)
+
+    def get_model_args(self):
+        model_args = super().get_model_args()
+        for key in [
+            "citation",
+            "in_samples",
+            "output_type",
+            "default_args",
+            "pred_sample",
+            "labels",
+            "sampling_rate",
+        ]:
+            del model_args[key]
+
+        model_args["in_channels"] = self.in_channels
+        model_args["classes"] = self.classes
+        model_args["phases"] = self.labels
+        model_args["sampling_rate"] = self.sampling_rate
+
+        return model_args
+
+    def get_defaults(self):
```

Comment:
Given `default_args` is a public variable, I think it's unnecessary adding a wrapper function.